tests/snappi/pfc/test_pfc_pause_response_with_snappi.py

```diff
@@ -0,0 +1,155 @@
+                 pause_prio_list=pause_prio_list,
+                 test_prio_list=test_prio_list,
+                 bg_prio_list=bg_prio_list,
+                 prio_dscp_map=prio_dscp_map,
+                 test_traffic_pause=True,
+                 headroom_test_params=headroom_test_params)
+
+
+def test_pfc_pause_multi_lossless_headroom(snappi_api,
+                                           snappi_testbed_config,
+                                           conn_graph_facts,
+                                           fanout_graph_facts,
+                                           duthosts,
+                                           rand_one_dut_hostname,
+                                           rand_one_dut_portname_oper_up,
+                                           lossless_prio_list,
+                                           lossy_prio_list,
+                                           prio_dscp_map,
+                                           enum_pfc_pause_delay_test_params):
+    """
+    Test headroom capacity for DUT for multiple lossless priorities
+
+    Args:
+        snappi_api (pytest fixture): SNAPPI session
+        snappi_testbed_config (pytest fixture): testbed configuration information
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+        duthosts (pytest fixture): list of DUTs
+        rand_one_dut_hostname (str): hostname of DUT
+        rand_one_dut_portname_oper_up (str): port to test, e.g., 's6100-1|Ethernet0'
+        lossless_prio_list (pytest fixture): list of all the lossless priorities
+        lossy_prio_list (pytest fixture): list of all the lossy priorities
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
+        enum_pfc_pause_delay_test_params (str): pfc delay value to test, 
+                                                  and delay responses e.g. "200|False"
+
+    Returns:
+        N/A
+    """
+
+    pytest_require(enum_pfc_pause_delay_test_params is not None, 
+                    "Skip this testcase since pfc pause delay values have not been configured yet")
+    dut_hostname, dut_port = rand_one_dut_portname_oper_up.split('|')
+    pytest_require(rand_one_dut_hostname == dut_hostname,
+                   "Port is not mapped to the expected DUT")
+
+    testbed_config, port_config_list = snappi_testbed_config
+    duthost = duthosts[rand_one_dut_hostname]
+    pause_prio_list = lossless_prio_list
+    test_prio_list = lossless_prio_list
+    bg_prio_list = lossy_prio_list
+
+    """ Populate headroom test params """
+    _, pfc_pause_delay_str, headroom_test_result_str = enum_pfc_pause_delay_test_params.split('|')
+    pfc_pause_delay = int(pfc_pause_delay_str)
+    headroom_test_result = True if headroom_test_result_str == 'True' else False
+    headroom_test_params = [pfc_pause_delay, headroom_test_result]
+    duthost.command("sonic-clear queuecounters")
```

Comment:
duthost.command should also be moved into run_pfc_test

Reply:
Done